File: frontend-ts/src/config/configLoader.ts
import defaultConfig from './config';

let profileConfig = {};

switch (process.env.REACT_APP_PROFILE) {
    case 'prod':
        profileConfig = require('./configProd').default;
        break;
    case 'test':
        profileConfig = require('./configTest').default;
        break;
    default:
        break;
}

const finalConfig = {
    ...defaultConfig,
    ...profileConfig
};

export default finalConfig;

File: backend/config_loader.py
import os

# Default configuration
from config import *

profile = os.environ.get('PROFILE')

if profile == 'prod':
    from config.config_prod import *
elif profile == 'test':
    from config.config_test import *
elif profile == 'remote-mongo':
    from config.config_test_remote_mongo import *
